### PR TITLE
feat(chat): implement isolated model view page (Issue #1154)

### PR DESCRIPTION
**Pull request by IlyaBondar**
**Description:**

Adding new page /models/:modelId to provide a possibility to open new sessions for selected models with just chat functionality.

Issues:

- Issue #1154

**Related PR**
- #1152 (author: @i-serebryakov07 )

**UI changes**

image

**Checklist:**

- [x] the pull request name complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] the pull request name starts with `fix(<scope>):`, `feat(<scope>):`, `feature(<scope>):`, `chore(<scope>):`, `hotfix(<scope>):` or `e2e(<scope>):`. If contains breaking changes then the pull request name must start with `fix(<scope>)!:`, `feat(<scope>)!:`, `feature(<scope>)!:`, `chore(<scope>)!:`, `hotfix(<scope>)!:` or `e2e(<scope>)!:` where `<scope>` is name of affected project: `chat`, `chat-e2e`, `overlay`, `shared`, `sandbox-overlay`, etc.
- [x] the pull request name ends with `(Issue #<TICKET_ID>)` (comma-separated list of issues)
- [x] I confirm that do not share any confidential information like API keys or any other secrets and private URLs

### CODE REVIEW

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1165.nightly-test.deltixhub.io
>E2E tests status: success

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1165.nightly-test.deltixhub.io
>E2E tests status: success

File: apps/chat/src/pages/index.tsx [hunk 1]
```diff
@@ -111,6 +113,13 @@ export default function Home({ initialState }: HomeProps) {
         }),
       );
     }
+    if (searchParams.has(ISOLATED_MODEL_QUERY_PARAM)) {
+      dispatch(
```
Comment (mikitabut) [inline apps/chat/src/pages/index.tsx]: set this setting state inside getServerSideProps

File: apps/chat/src/pages/index.tsx [hunk 2]
```diff
@@ -229,9 +238,8 @@ export default function Home({ initialState }: HomeProps) {
             <div className="flex size-full flex-col sm:pt-0">
               {enabledFeatures.has(Feature.Header) && <Header />}
               <div className="flex w-full grow overflow-auto">
-                {enabledFeatures.has(Feature.ConversationsSection) && (
-                  <Chatbar />
-                )}
+                {enabledFeatures.has(Feature.ConversationsSection) &&
```
Comment (mikitabut) [inline apps/chat/src/pages/index.tsx]: Maybe it's will be easier to just set in getServerSideProps needed features than add additional ifs on all levels of rendering?

**Comment by mikitabut** [comment]
Please add documentation about that in our readme

File: apps/chat/src/pages/index.tsx (hunk 1 above)
Comment (IlyaBondar) [reply apps/chat/src/pages/index.tsx]: @mikitabut , `window is not defined` into `getServerSideProps`

File: apps/chat/src/pages/index.tsx (hunk 2 above)
Comment (IlyaBondar) [reply apps/chat/src/pages/index.tsx]: @mikitabut , `window is not defined` into `getServerSideProps`

File: apps/chat/src/components/Header/Header.tsx [hunk 3]
```diff
@@ -79,36 +83,38 @@ const Header = () => {
...
-            />
-          )}
-        </div>
-      </Tooltip>
-      <CreateNewChatMobile iconSize={headerIconSize} />
+            )}
+          </div>
+        </Tooltip>
+      )}
+      {!isIsolatedView && <CreateNewChatMobile iconSize={headerIconSize} />}
```
Comment (mikitabut) [inline apps/chat/src/components/Header/Header.tsx]: maybe put this button under feature flag and disable it in same manner as conversations section and etc?

**Comment by IlyaBondar** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-chat-pr-1165.nightly-test.deltixhub.io
>E2E tests status: success

**Review by denys-kolomiitsev — APPROVED** [review]

**Pull request merged**